### PR TITLE
Accept pathlib.Path as arguments

### PR DESCRIPTION
**Pull request by pykong**
This **PR** modernizes `pathspec` to work with pathlib.Path objects.

- modified `normalize_file` and `normalize_files`
- added tests
- updated docstrings

Closes: #33

### CODE REVIEW

File: pathspec/tests/test_util.py [hunk 1]
```diff
@@ -367,3 +368,24 @@ def test_3_entries(self):
...
+		"""
+		Tests passing pathlib.Path as argument.
+		"""
+		from pathlib import Path
+		first_spec = normalize_file(Path('a.txt'))
+		second_spec = normalize_file('a.txt')
+		self.assertEqual(first_spec, second_spec)
+
+	@unittest.skipIf(sys.version_info < (3, 4), "pathlib entered stdlib in Python 3.4")
+	def test_4_normalizing_pathlib_paths(self):
```
Comment (cpburnz) [inline pathspec/tests/test_util.py]: Due to my comment on *util.normalize_files()* this test won't be needed.

File: pathspec/util.py [hunk 2]
```diff
@@ -302,7 +302,7 @@ def normalize_files(files, separators=None):
 	"""
 	norm_files = {}
 	for path in files:
-		norm_files[normalize_file(path, separators=separators)] = path
+		norm_files[normalize_file(path, separators=separators)] = str(path)
```
[the remark this answers is not in the document]
Comment (pykong) [reply pathspec/util.py]: Done via 77521fe

File: pathspec/pathspec.py [hunk 3]
```diff
@@ -132,6 +133,7 @@ def match_tree_files(self, root, on_error=None, follow_links=None):
 		Returns the matched files (:class:`~collections.abc.Iterable` of
 		:class:`str`).
 		"""
+		root = str(root)  # stringify pathlib.Path
```
[the remark this answers is not in the document]
Comment (pykong) [reply pathspec/pathspec.py]: Done via 77521fe

File: pathspec/tests/test_util.py (hunk 1 above)
Comment (pykong) [reply pathspec/tests/test_util.py]: Done via 77521fe

**Comment by pykong** [comment]
@cpburnz Thanks for having a look. 
I have implemented the requested changes.

**Comment by cpburnz** [comment]
Looks good, thanks.

**Comment by pykong** [comment]
You are welcome.
Btw - this is one library making very good use of `pathspec`:
https://github.com/pykong/copier#patterns-syntax

**Pull request merged**